diff --git a/README.md b/README.md
index 0ebab3f..cbc8532 100644
--- a/README.md
+++ b/README.md
@@ -8,6 +8,7 @@ Uses powerline to setup bash, vim, and tmux.
 Tested distributions:
  - [ ] CentOS 7.x
  - [x] CentOS 8.x
+ - [x] Debian 10
  - [ ] Red Hat Enterprise Linux 7.x
  - [x] Red Hat Enterprise Linux 8.3
  - [ ] Red Hat Enterprise Linux 8.4
@@ -18,4 +19,4 @@ Tested distributions:
 
 ToDo:
  - [ ] Test with system-wide powerline installation
- - [ ] RHEL 8.4 uses python3.8 for python3 but vim uses python3.6, so installing powerline module is not visible to vim from python3.8 site-packages.  Workaround is to also install powerline with the command "python3.6 -m pip install --user powerline-status powerline-gitstatus".  Figure out how to test with setup.sh script.
+ - [x] RHEL 8.4 uses python3.8 for python3 but vim uses python3.6, so installing powerline module is not visible to vim from python3.8 site-packages.  Workaround is to also install powerline with the command "python3.6 -m pip install --user powerline-status powerline-gitstatus".  Figure out how to test with setup.sh script.
diff --git a/setup.sh b/setup.sh
index c6ec316..0307411 100755
--- a/setup.sh
+++ b/setup.sh
@@ -53,6 +53,21 @@ else
 fi
 echo -e "${INFO} Found $OS $VER [$ID]"
 
+if [ $ID == "centos" ]; then
+  :
+elif [ $ID == "rhel" ]; then
+  :
+elif [ $ID == "fedora" ]; then
+  :
+elif [ $ID == "ubuntu" ]; then
+  :
+elif [ $ID == "debian" ]; then
+  :
+else
+  echo -e "${CROSS} Validating OS distro failed, $SCRIPT_NAME must be modified"
+  exit 1
+fi
+
 
 CURL_CHECK="Checking for curl"
 # Check for important packages
@@ -67,7 +82,18 @@ fi
 VIM_CHECK="Checking for vim"
 if ! command -v vim &> /dev/null; then
   echo -e "${CROSS} $VIM_CHECK"
-  PACKAGES+=("vim")
+  if [ $ID == "centos" ]; then
+    PACKAGES+=("vim")
+  elif [ $ID == "rhel" ]; then
+    PACKAGES+=("vim")
+  elif [ $ID == "fedora" ]; then
+    PACKAGES+=("vim")
+  elif [ $ID == "ubuntu" ]; then
+    PACKAGES+=("vim")
+  elif [ $ID == "debian" ]; then
+    PACKAGES+=("vim-nox")
+  fi
+  # ToDo: What to do for other distros??
 else
   echo -e "${TICK} $VIM_CHECK"
 fi
@@ -83,6 +109,8 @@ if ! command -v ctags &> /dev/null; then
     PACKAGES+=("ctags")
   elif [ $ID == "ubuntu" ]; then
     PACKAGES+=("exuberant-ctags")
+  elif [ $ID == "debian" ]; then
+    PACKAGES+=("exuberant-ctags")
   fi
   # ToDo: What to do for other distros??
 else
@@ -127,6 +155,9 @@ if (( ${#PACKAGES[@]} > 0 )); then
   elif [ $ID == "ubuntu" ]; then
     echo -e "${INFO} Run 'sudo apt install -y ${PACKAGES[@]}'"
     exit 1
+  elif [ $ID == "debian" ]; then
+    echo -e "${INFO} Run 'sudo apt install -y ${PACKAGES[@]}'"
+    exit 1
   fi
   echo -e "${CROSS} Setup incomplete, resolve dependencies and run setup again"
   exit 1
@@ -169,7 +200,7 @@ if [[ ! -z $VIM_PYTHON_VER && $SYS_PYTHON_VER != $VIM_PYTHON_VER ]]; then
 fi
 
 # Ensure we have a ~/.config directory
-[ ! -d "$HOME/.config" ] && mkdir -p "$HOME/.config"
+[ ! -d "$HOME/.config" ] && mkdir -p "$HOME/.config" && echo -e "${TICK} Creating $HOME/.config"
 
 # Backup any existing powerline configuration
 POWERLINE_BACKUP_INSTALL="Backing up existing powerline config file"
@@ -180,7 +211,7 @@ elif [ -L "$HOME/.config/powerline.backup" ]; then
   # echo -e "${CROSS} $POWERLINE_BACKUP_INSTALL skipped, found existing symlink"
   :
 else
-  [ -d "$HOME/.config/powerline" ] && mv "$HOME/.config/powerline" "$HOME/.config/powerline.backup" && echo -e "${CHECK} $POWERLINE_BACKUP_INSTALL"
+  [ -d "$HOME/.config/powerline" ] && mv "$HOME/.config/powerline" "$HOME/.config/powerline.backup" && echo -e "${TICK} $POWERLINE_BACKUP_INSTALL"
 fi
 
 # Symlink powerline configuration from git source directory

